test/addons-napi/test_async/test_async.cc

```diff
@@ -0,0 +1,130 @@
+#include <node_api.h>
+
+#if defined _WIN32
+#include <windows.h>
+#else
+#include <unistd.h>
+#endif
+
+#define NAPI_CHECK(env, condition, msg)                                   \
+  if (!(condition)) {                                                     \
+    napi_throw_error((env), (msg));                                       \
+    return;                                                               \
+  }
+
+#define NAPI_CALL(env, theCall)                                           \
+  if ((theCall) != napi_ok) {                                             \
+    const napi_extended_error_info* error;                                \
+    napi_get_last_error_info((env), &error);                              \
+    const char *errorMessage = error->error_message;                      \
+    errorMessage = errorMessage ? errorMessage : "empty error message";   \
```

Comment:
Good point. This will be refactored via https://github.com/nodejs/node/pull/12248 so I didn't pull all of that in for this change. Will make this change, though.